src/main/java/com/example/benefit/service/BenefitService.java

```diff
@@ -35,10 +39,19 @@ public BenefitDTO createAndUpdateBenefit(BenefitDTO benefitDTO) {
             benefit.setCalculationMethod(calculationMethodRepository.findByName(calculationMethodName));
         }
 
+        BenefitDTOForRabbitMQ benefitDTOForRabbitMQ = BenefitDTOForRabbitMQ.builder()
+                .id(benefitDTO.getId())
+                .name(benefitDTO.getName())
+                .benefitTypeName(benefitDTO.getBenefitTypeDTO().getName())
+                .calculationMethodName(benefitDTO.getCalculationMethodDTO().getName())
+                .build();
+
         if (benefitRepository.findByName(benefit.getName()) == null) {
             benefitRepository.save(benefit);
+            streamBridge.send("benefit-out-0",benefitDTOForRabbitMQ);
             return benefitMapper.entityToDto(benefit);
         } else {
+            streamBridge.send("benefit-out-0",benefitDTOForRabbitMQ);
             return benefitMapper.entityToDto(benefitRepository.findByName(benefit.getName()));
```

Comment:
We don't really update existing benefits, let's fix it, for reference we can use Employee service